app/enterprise/1.3-x/developer-portal/theme-customization/single-page-app.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Hosting Single Page App out of Kong Dev Portal
+---
+
+### Introduction
+
+The Kong Developer Portal ships with a default server side rendered theme, however it is possible to replace this with a Single Page App(SPA). Our example is in Angular using the Angular CLI Tool, but you can follow along with any other javascript framework, with just a few tweaks.
+
+To view the basic example Angular template from this this guide view the [`example/spa-angular` branch from the kong-portal-templates branch](https://github.com/Kong/kong-portal-templates/tree/example/spa-angular).
+
+### Prerequisites
+
+* Kong Enterprise 1.3 or later
+* Portal Legacy is turned off
+* The Developer Portal is enabled and running
+* kong-portal-cli tool is installed locally
+
+### What is a SPA
+
+A Single Page App(SPA) is a website that loads all the HTML, Javascript, and CSS on the first load. Instead of loading subsequent pages from the server, javascript is used to dynamically change the page content. You may wish to use a SPA in Dev Portal if you have a preexisting SPA you want to integrate with portal, or you are trying to achieve a more application like experience across many pages. A SPA takes control of routing from the server, and handles it client side instead.
+
+If you simplify wish to add interactive javascript to a few pages, it may make more sense to add the javascript to only layouts that require it, and maintain server side rendering. [Learn more about adding Javascipt to a layout without implementing a SPA.](/enterprise/{{page.kong_version}}/developer-portal/theme-customization/adding-javascript-assets)
+
+### Making Choices
+
+Decide what pages routing should be handled by SPA and what is handled by content files.
+
+We recommend Catalog and Spec routes not be handled by SPA
+If you are using Authentication, then you probably also want to leave server-side rendering for any account pages
+
+### Getting Started
+1. Clone down the portal-templates repo
+
+2. Create a file called `router.conf.yaml` in `workspaces/default` This file will override the default routing, allowing you to control routing via javascript.
+
+`router.conf.yaml` must be a yaml file, where the key is each route, and the value a content or spec path. `/*` Is a catch-all wildcard for all routes not specified in `router.conf.yaml`, it will overwrite all default routing from content/spec path or set in headmatter.
```

Comment:
```suggestion
`router.conf.yaml` must be a yaml file, where the key is each route, and the value a content or spec path. `/*` Is a catch-all wildcard for all routes not specified in `router.conf.yaml`, it will overwrite all default routing set by collections or set in headmatter.
```